openssl/src/symm.rs

```diff
@@ -623,10 +678,21 @@ pub fn decrypt_aead(
 ) -> Result<Vec<u8>, ErrorStack> {
     let mut c = Crypter::new(t, Mode::Decrypt, key, iv)?;
     let mut out = vec![0; data.len() + t.block_size()];
+
+    if t.is_ccm() {
+        c.set_tag(tag)?;
+        c.set_data_len(data.len())?;
+    }
+
     c.aad_update(aad)?;
     let count = c.update(data, &mut out)?;
-    c.set_tag(tag)?;
-    let rest = c.finalize(&mut out[count..])?;
+    let mut rest = 0;
+
+    if !t.is_ccm() {
+        c.set_tag(tag)?;
+        rest = c.finalize(&mut out[count..])?;
```

Comment:
finalize doesn't need to be called at all with CCM?

Reply:
I'm not entirely sure. The documentation doesn't state that finalize is mandatory for ccm and [the example that ships with openssl](https://github.com/openssl/openssl/blob/master/demos/evp/aesccm.c#L87-L119) skips it during decryption as well.

Reply:
Interesting. I guess that makes sense given that you can only call update once. Can you add a test making sure that a bad tag properly returns an error? That's the one thing I could imagine finalize doing for CCM.